concrete/src/Support/CodingStyle/PhpFixer.php

```diff
@@ -0,0 +1,266 @@
+     */
+    protected $runner;
+
+    /**
+     * Initialize the instance.
+     *
+     * @param \Concrete\Core\Support\CodingStyle\PhpFixerOptions $options
+     * @param \Concrete\Core\Support\CodingStyle\PhpFixerRuleResolver $ruleResolver
+     * @param \Concrete\Core\Support\CodingStyle\PhpFixerRunner $runner
+     */
+    public function __construct(PhpFixerOptions $options, PhpFixerRuleResolver $ruleResolver, PhpFixerRunner $runner)
+    {
+        $this->options = $options;
+        $this->ruleResolver = $ruleResolver;
+        $this->runner = $runner;
+    }
+
+    /**
+     * Get the fixer options.
+     *
+     * @return \Concrete\Core\Support\CodingStyle\PhpFixerOptions
+     */
+    public function getOptions()
+    {
+        return $this->options;
+    }
+
+    /**
+     * @param \Symfony\Component\Console\Input\InputInterface $input
+     * @param \Symfony\Component\Console\Output\OutputInterface $output
+     * @param \SplFileInfo[] $paths with absolute paths
+     * @param bool $dryRun
+     *
+     * @return array
+     */
+    public function fix(InputInterface $input, OutputInterface $output, array $paths, $dryRun = false)
+    {
+        $pathRuleList = [];
+        foreach ($paths as $path) {
+            $pathRuleList = $this->mergePathAndFlags($pathRuleList, $this->splitPathToPathAndFlags($path));
+        }
+        $this->runner->resetSteps();
+        foreach ($pathRuleList as $flags => $paths) {
+            $this->runner->addStep($this->options, $paths, $flags);
+        }
+
+        $progressOutput = new ProcessOutput($output, $this->runner->getEventDispatcher(), $this->runner->calculateNumberOfFiles());
+        $counters = [];
+        $counter = function (FixerFileProcessedEvent $e) use (&$counters) {
+            $status = $e->getStatus();
+            if (isset($counters[$status])) {
+                ++$counters[$status];
+            } else {
+                $counters[$status] = 1;
+            }
+        };
+        $this->runner->getEventDispatcher()->addListener(FixerFileProcessedEvent::NAME, $counter);
+        try {
+            $progressOutput->printLegend();
+            $changes = $this->runner->apply($dryRun);
+            $output->writeln('');
+        } finally {
+            $this->runner->getEventDispatcher()->removeListener(FixerFileProcessedEvent::NAME, $counter);
+        }
+
+        return [$counters, $changes, clone $this->runner->getErrorManager()];
+    }
+
+    /**
+     * @param \SplFileInfo $path
+     *
+     * @throws \RuntimeException if $path could not be found
+     *
+     * @return array
+     */
+    protected function splitPathToPathAndFlags(SplFileInfo $path)
+    {
+        if ($path->isFile()) {
+            return $this->splitFileToPathAndFlags($path);
+        }
+        if ($path->isDir()) {
+            return $this->splitDirectoryToPathAndFlags($path);
+        }
+        throw new RuntimeException(t('Failed to find the file/directory %s', $path->getPathname()));
```

Comment:
Do translations make sense if this is called via CLI?

Reply:
Nope, but we don't know if someone will make an add-on that uses this class.